src/Models/PathEditModel.vala

```diff
@@ -117,17 +128,171 @@ public class Akira.Models.PathEditModel : Object {
+            }
+        }
+
+        return Lib.Modes.PathEditMode.PointType.NONE;
+    }
+
+    /*
+     * This method will be used when editing paths to update the position of a point.
+     */
+    public void modify_point_value (Geometry.Point new_pos) {
+        new_pos.x -= first_point.x;
+        new_pos.y -= first_point.y;
+
+        if (selected_pts.size == 1) {
+            // If only one point is selected, it will act as reference point.
+            reference_point = selected_pts.to_array ()[0];
+        }
+
+        Geometry.Point original_ref = Geometry.Point (points[reference_point].x, points[reference_point].y);
+
+        // Calculate by how much the reference point changed, then move other points by this amount.
+        double delta_x = original_ref.x - new_pos.x;
+        double delta_y = original_ref.y - new_pos.y;
+
+        if (tangents_inline) {
+            // If a tangent line was selected and they were inline,
+            // they need to be moved in a special way.
+            move_tangents_rel_to_curve (delta_x, delta_y);
+        } else {
+            foreach (var item in selected_pts) {
+                points[item].x -= delta_x;
+                points[item].y -= delta_y;
+            }
+        }
+
+        points = recalculate_points (points);
+        bool close = instance.components.path.close;
+        instance.components.path = new Lib.Components.Path.from_points (points, commands, close);
+        recompute_components ();
+    }
+
+    /*
+     * This method is used to change the selected points.
+     * idx is index of point to be added to selection. if -1, it is not added.
+     * If append is true, add the item to the list, otherwise, clear the list then add.
+     */
+    public void set_selected_points (int idx, bool append = false) {
+        if (!append) {
+            selected_pts.clear ();
+        }
+
+        if (idx != -1) {
+            selected_pts.add (idx);
+        }
+
+        update_view ();
+    }
+
+    /*
+     * This method is used to join the first and last point of path
+     * to make it a closed curve.
+     */
+    public void make_path_closed () {
+        instance.components.path = new Lib.Components.Path.from_points (points, commands, true);
+        recompute_components ();
+    }
+
+    public bool check_can_path_close () {
+        int last = points.length - 1;
+
+        if (last == 0) {
+            return false;
+        }
+
+        double delta_x = (points[0].x - points[last].x).abs ();
+        double delta_y = (points[0].y - points[last].y).abs ();
+
+        if (delta_x <= 2 && delta_y <= 2) {
+            make_path_closed ();
+            return true;
+        }
+
+        return false;
+    }
+
+    private bool compare_points (Geometry.Point a, Geometry.Point b) {
+        double thresh = 4 / view_canvas.scale;
+        double delta_x = Math.ceil ((a.x + first_point.x - b.x).abs ());
+        double delta_y = Math.ceil ((a.y + first_point.y - b.y).abs ());
```

Comment:
Should just use distance from geometry math or similar. If you use magic numbers (4), make them a private static int and name it so that people know what it is.